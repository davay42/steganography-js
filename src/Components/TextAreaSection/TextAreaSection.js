import React, { Component } from "react";
import './TextAreaSection.css';

class CanvasSection extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    resetState() {
        this._textArea.value = "";
    }

    setText(text) {
        this._textArea.value = text;
    }

    handleProcess() {
        // Get the message string
        let string = JSON.stringify(this._textArea.value);
        let uint8_array = new TextEncoder(/* document.characterSet.toLowerCase() */).encode(string);
        let payload = uint8_array.buffer;
        this.props.process({
            process: this.props.processName,
            text: {
                buffer: payload,
                length: string.length
            }
        })
    }

    render() {
        const disableButtons =
            !this.props.sourceFileLoaded
            || this.props.isAProcessActive;

        return (
            <div>
                <br />
                <div className="section-content input-field">
                    <label className="label"
                        htmlFor={this.props.id + "-text"}>
                        Text I/O *
                    </label>
                    <br /><br />
                    <textarea
                        className="materialize-textarea textarea"
                        id={this.props.id + "-text"}
                        ref={(ref) => this._textArea = ref}>
                    </textarea>
                    <span>* <small>Larger text output will be automatically downloaded as a .txt file. These files can be several MBs in size depending on the source image, so try opening it with a roboust text editor.</small></span>
                </div>
                <div className="section-actions secondary">
                    <button
                        className="waves-effect waves-light btn yellow darken-1 black-text"
                        disabled={this.props.isAProcessActive}
                        onClick={() => this.resetState()}>
                        Clear
                    </button>
                </div>
                <div className="section-actions">
                    <button
                        className="waves-effect waves-light btn purple darken-4"
                        disabled={disableButtons}
                        onClick={() => this.handleProcess()}>
                        {this.props.processName}
                    </button>
                </div>
            </div>
        )
    }
}

export default CanvasSection;
