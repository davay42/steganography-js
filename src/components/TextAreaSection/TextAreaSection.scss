.TextAreaSelection .textarea {
    font-size: 1em !important;
    font-family: monospace;
    height: 18em !important;
    min-width: 20em !important;
    overflow-x: hidden;
    overflow-y: scroll !important;
}

.TextAreaSelection textarea.materialize-textarea:focus:not([readonly]) {
    border-bottom: 1px solid #9c27b0;
    -webkit-box-shadow: 0 1px 0 0 #9c27b0;
    box-shadow: 0 1px 0 0 #9c27b0;
}
