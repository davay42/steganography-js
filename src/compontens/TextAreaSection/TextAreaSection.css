.textarea {
  font-size: 1em !important;
  font-family: monospace;
  height: 10em !important;
  min-width: 20em !important;
  overflow-x: hidden;
  overflow-y: scroll !important;
}

textarea.materialize-textarea:focus:not([readonly]) {
  border-bottom: 1px solid #9c27b0;
  -webkit-box-shadow: 0 1px 0 0 #9c27b0;
  box-shadow: 0 1px 0 0 #9c27b0;
}