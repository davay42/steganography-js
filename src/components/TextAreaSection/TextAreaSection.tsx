import React, { Component, createRef } from "react";
import LangContext from "../../context/LangContext";
import "./TextAreaSection.scss";

interface Props {
    clear?: any;
    id?: string;
    isAProcessActive?: boolean;
    sourceFileLoaded?: boolean;
    processName?: ProcessType;
    onFileLoaded?(bool: boolean, fileName: string): void;
    process?(args: { process: string, text: TextPayload }): void;
}

export default class TextAreaSection extends Component<Props> {
    _textArea = createRef<HTMLTextAreaElement>();

    resetState = () => {
        if (this._textArea.current) {
            this._textArea.current.value = "";
        }
    }

    setText = (text: string) => {
        if (this._textArea.current) {
            this._textArea.current.value = text;
        }
    }

    handleProcess = () => {
        const string = JSON.stringify(this._textArea.current?.value);
        const uint8_array = new TextEncoder().encode(string);
        const payload = uint8_array.buffer;

        if (this.props.process && this.props.processName) {
            this.props.process({
                process: this.props.processName,
                text: {
                    buffer: payload,
                    length: string.length,
                },
            });
        }
    }

    render() {
        const t = this.context;
        const disableButtons = !this.props.sourceFileLoaded ||
            this.props.isAProcessActive;

        return (
            <div className="TextAreaSelection">
                <div className="section-content input-field">
                    <label className="label" htmlFor={this.props.id + "-text"}>
                        {t("text_input:label")} *
                    </label>
                    <br />
                    <br />
                    <textarea
                        className="materialize-textarea textarea"
                        id={this.props.id + "-text"}
                        ref={this._textArea}
                    >
                    </textarea>
                    <span>
                        * <small>{t("text_input:large_file_info")}</small>
                    </span>
                </div>
                <div className="section-actions secondary">
                    <button
                        className="waves-effect waves-light btn pink darken-1 black-text"
                        disabled={this.props.isAProcessActive}
                        onClick={() => this.resetState()}
                    >
                        {t("common:clear")}
                    </button>
                </div>
                <div className="section-actions">
                    <button
                        className="waves-effect waves-light btn grey darken-4"
                        disabled={disableButtons}
                        onClick={() => this.handleProcess()}
                    >
                        {this.props.processName}
                    </button>
                </div>
            </div>
        );
    }
}

TextAreaSection.contextType = LangContext;
