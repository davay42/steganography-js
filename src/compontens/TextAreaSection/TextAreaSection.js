import React, { Component } from "react";
import LangContext from "../../context/LangContext";
import "./TextAreaSection.scss";

export default class TextAreaSection extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    resetState() {
        this._textArea.value = "";
    }

    setText(text) {
        this._textArea.value = text;
    }

    handleProcess() {
        const string = JSON.stringify(this._textArea.value);
        const uint8_array = new TextEncoder().encode(string);
        const payload = uint8_array.buffer;
        this.props.process({
            process: this.props.processName,
            text: {
                buffer: payload,
                length: string.length,
            },
        });
    }

    render() {
        const t = this.context;
        const disableButtons = !this.props.sourceFileLoaded ||
            this.props.isAProcessActive;

        return (
            <div className="TextAreaSelection">
                <div className="section-content input-field">
                    <label className="label" htmlFor={this.props.id + "-text"}>
                        {t("text_input:label")} *
                    </label>
                    <br />
                    <br />
                    <textarea
                        className="materialize-textarea textarea"
                        id={this.props.id + "-text"}
                        ref={(ref) => this._textArea = ref}
                    >
                    </textarea>
                    <span>
                        * <small>{t("text_input:large_file_info")}</small>
                    </span>
                </div>
                <div className="section-actions secondary">
                    <button
                        className="waves-effect waves-light btn pink darken-1 black-text"
                        disabled={this.props.isAProcessActive}
                        onClick={() => this.resetState()}
                    >
                        {t("common:clear")}
                    </button>
                </div>
                <div className="section-actions">
                    <button
                        className="waves-effect waves-light btn grey darken-4"
                        disabled={disableButtons}
                        onClick={() => this.handleProcess()}
                    >
                        {this.props.processName}
                    </button>
                </div>
            </div>
        );
    }
}

TextAreaSection.contextType = LangContext;
