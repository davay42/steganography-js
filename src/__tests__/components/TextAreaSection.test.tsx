import React from 'react';
import renderer from 'react-test-renderer';

import TextAreaSection from '../../components/TextAreaSection/TextAreaSection';

it('renders CanvasSection', () => {
    const tree = renderer
        .create(<TextAreaSection />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
