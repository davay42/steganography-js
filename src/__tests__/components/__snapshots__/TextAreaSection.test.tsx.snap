// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CanvasSection 1`] = `
<div
  className="TextAreaSelection"
>
  <div
    className="section-content input-field"
  >
    <label
      className="label"
      htmlFor="undefined-text"
    >
      {text_input:label}
       *
    </label>
    <br />
    <br />
    <textarea
      className="materialize-textarea textarea"
      id="undefined-text"
    />
    <span>
      * 
      <small>
        {text_input:large_file_info}
      </small>
    </span>
  </div>
  <div
    className="section-actions secondary"
  >
    <button
      className="waves-effect waves-light btn pink darken-1 black-text"
      onClick={[Function]}
    >
      {common:clear}
    </button>
  </div>
  <div
    className="section-actions"
  >
    <button
      className="waves-effect waves-light btn grey darken-4"
      disabled={true}
      onClick={[Function]}
    />
  </div>
</div>
`;
